cli/cmd/options.go

```diff
@@ -63,19 +57,39 @@ func makeInstallUpgradeFlags(defaults *l5dcharts.Values) ([]flag.Flag, *pflag.Fl
 						return err
 					}
 					// The HA flag must be processed before flags that set these values individually so that the
-					// individual flags can override the HA defaults.  This means that the HA flag must appear
-					// before the individual flags in the slice passed to flags.ApplyIfSet.
-					values.ControllerReplicas = haValues.ControllerReplicas
+					// individual flags can override the HA defaults.
+					values.EnablePodAntiAffinity = haValues.EnablePodAntiAffinity
 					values.GetGlobal().Proxy.Resources.CPU.Request = haValues.GetGlobal().Proxy.Resources.CPU.Request
 					values.GetGlobal().Proxy.Resources.Memory.Request = haValues.GetGlobal().Proxy.Resources.Memory.Request
-					// NOTE: CPU Limits are not currently set by default HA charts.
-					//values.Global.Proxy.Cores = haValues.Global.Proxy.Cores
-					//values.Global.Proxy.Resources.CPU.Limit = haValues.Global.Proxy.Resources.CPU.Limit
 					values.GetGlobal().Proxy.Resources.Memory.Limit = haValues.GetGlobal().Proxy.Resources.Memory.Limit
+					values.ControllerReplicas = haValues.ControllerReplicas
+					values.DestinationResources = haValues.DestinationResources
+					values.PublicAPIResources = haValues.PublicAPIResources
+					values.IdentityResources = haValues.IdentityResources
+
+					initResourceMaps(values.Grafana, haValues.Grafana)
+					values.Grafana["resources"] = haValues.Grafana["resources"]
+
+					values.HeartbeatResources = haValues.HeartbeatResources
+
+					initResourceMaps(values.Prometheus, haValues.Prometheus)
+					values.Prometheus["resources"] = haValues.Prometheus["resources"]
+
+					values.ProxyInjectorResources = haValues.ProxyInjectorResources
+					values.WebhookFailurePolicy = haValues.WebhookFailurePolicy
+					values.SPValidatorResources = haValues.SPValidatorResources
+					values.TapResources = haValues.TapResources
+					values.WebResources = haValues.WebResources
```

Comment:
Instead of individually merging these values, what are your thoughts on using `func (v Values) merge(src Values) (Values, error)`?

It needs to be made public, but it takes care of this and we don't need to worry about keeping this in sync with HA values if they are ever added or removed.

I tried this locally already and it seems to work fine. The only differences I see in the `linkerd install --ha` output (besides git version and TLS related values) are:

- The absence of the `Resources: {}` field that I mentioned in another comment
- The `linkerd-config` has some tracing values that I don't see in the output of your PR

Reply:
Good call, I've pushed a change using `Merge()` instead. I couldn't find though the tracing differences you mentioned, could you clarify?

Reply:
Looks like they somehow got there in quickly testing in `Merge()` worked for me. I don't see them now 👍